packages/react-sdk/src/state/communication/matrixRtcCommunicationChannel.ts

```diff
@@ -184,116 +184,118 @@ export class MatrixRtcCommunicationChannel implements CommunicationChannel {
     this.peerConnections.forEach((c) => c.close());
   }
 
-  private async initLiveKitServer(session: Session): Promise<void> {
-    if (this.sfuConfig !== undefined) {
-      this.logger.debug('LiveKit config already set, skipping init');
-      // LiveKit servers already known
-      return;
+  private async initFocusBackend(focus: RTCFocus): Promise<void> {
+    this.logger.debug('Initializing focus backend with focus', focus);
+
+    if (focus.type !== 'livekit') {
+      this.logger.error(
+        'Unable to init focus backend due to unsupported active focus type',
+        focus.type,
+      );
+      throw new Error('Unsupported active focus type');
+    }
+
+    if (!isLivekitFocusConfig(focus)) {
+      this.logger.error(
+        'Unable to init focus backend due to unsupported LiveKit configuration',
+        focus,
+      );
+      throw new Error('Unsupported active focus backend configuration');
     }
 
-    try {
-      const widgetApi = await this.widgetApiPromise;
+    const widgetApi = await this.widgetApiPromise;
+    if (!widgetApi.widgetParameters.roomId) {
+      this.logger.error('Room ID not found in widget parameters');
+      throw new Error('Room ID not found in widget parameters');
+    }
 
-      if (!widgetApi.widgetParameters.roomId) {
-        this.logger.error('Room ID not found in widget parameters');
-        throw new Error('Room ID not found in widget parameters');
-      }
+    // Disconnect before starting a new backend connection
+    await this.disconnect();
 
-      const focus = (
-        await makePreferredLivekitFoci(
-          session,
-          widgetApi.widgetParameters.roomId,
-        )
-      )[0];
-      this.sfuConfig = await getSFUConfigWithOpenID(widgetApi, focus);
+    // The LiveKit alias is required to check for room membership when requesting the
+    // OpenID token and place users in the same LiveKit data exchange room
+    const activeFocus: LivekitFocus = {
+      ...focus,
+      livekit_alias: widgetApi.widgetParameters.roomId,
+    };
 
-      this.logger.debug('Got SFU config', JSON.stringify(this.sfuConfig));
+    this.sfuConfig = await AutoDiscovery.getSFUConfigWithOpenID(
+      widgetApi,
+      activeFocus,
+    );
 
-      if (!this.sfuConfig) {
-        this.logger.warn('Failed to get LiveKit JWT');
-        throw new Error('Failed to get LiveKit JWT');
-      }
-    } catch (error) {
-      this.logger.error('Error getting the LiveKit SFU config', error);
+    if (!this.sfuConfig) {
+      this.logger.error('Unable to retrieve LiveKit SFU configuration');
+      throw new Error('Unable to retrieve LiveKit SFU configuration');
     }
-  }
 
-  private async handleSessionJoined(session: Session): Promise<void> {
-    const sessionId = this.sessionManager.getSessionId();
+    const { userId, deviceId } = widgetApi.widgetParameters;
+    const sessionId = `_${userId}_${deviceId}`;
+    const session = { sessionId: sessionId, userId: userId } as Session;
```

Comment:
```suggestion
    const session: Session = { sessionId: sessionId, userId: userId };
```
would also require `userId` to be checked for undefined